examples/pxScene2d/src/rcvrcore/XModule.js

```diff
@@ -28,63 +27,24 @@ function XModule(name, appSceneContext, basePath, jarName) {
   this.name = name;
   this.appSandbox = null;
   this.moduleReadyPromise = null;
-  this.exports = {};
-  this.pendingIncludes = {};
-  this.moduleNameList = [];
-  this.promises = [];
-  this.moduleData = {};
+  this.exports = {}; // module.exports is an object by default
   this.appSceneContext = appSceneContext;
-  this.imports = {};
   this.log = null;
   this.basePath = basePath;
   this.jarName = jarName;
   this.importReplacementMap = {};
 }
 
-XModule.prototype.load = function(uri) {
-  global.exports = self.exports;
-  global.module = self;
-};
-
 XModule.prototype.freeResources = function() {
   this.name = null;
   this.appSandbox = null;
   this.moduleReadyPromise = null;
-  for(var key in this.exports) {
-    this.exports[key] = null;
-    delete this.exports[key];
-  }
   this.exports = null;
-  for(var key in this.pendingIncludes) {
-    this.pendingIncludes[key] = null;
-    delete this.pendingIncludes[key];
-  }
-  this.pendingIncludes = null;
-  if ((undefined != this.moduleNameList) && (null != this.moduleNameList))
-  {
-    var nmodules = this.moduleNameList.length;
-    for (var i=0; i<nmodules; i++)
-    {
-      this.moduleNameList.pop();
-    }
-  }
-  this.moduleNameList = null;
-  if ((undefined != this.promises) && (null != this.promises))
-  {
-    var npromises = this.promises.length;
-    for (var i=0; i<npromises; i++)
-    {
-      this.promises.pop();
-    }
-  }
-  this.promises = null;
-  this.moduleData = null;
   this.appSceneContext = null;
-  this.imports = null;
   this.log = null;
-  this.importReplacementMap = null;
```

Comment:
Can you please briefly describe why we no longer want to clean up these references inside of freeResources?

Reply:
imports, moduleData, promises, moduleNameList, pendingIncludes have been removed. As to 'exports', '= null' is sufficient in javascript.